### ISSUE TITLE
[data grid] Inaccessible Export menu in toolbar

### ISSUE THREAD

**Issue by nogalpaulina** [mui/mui-x#5476]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

The Export button within the MUI Data Grid Toolbar opens a menu with export options (print, CSV). These options aren’t accessible for assistive technology users as they can’t access the menu with a screen reader on, or by tabbing. The reason why is the tabindex=”-1” applied to the menu (MuiList) and list items (MuiMenuItem). Developers don’t have enough control over this component to fix it so they need to develop alternative ways to provide the same Export options for assistive technology users. 

<img width="1058" alt="Screen Shot 2022-07-08 at 3 12 15 PM" src="https://user-images.githubusercontent.com/10341961/178547465-199e07a5-d234-43f8-80f5-613b51ccded0.png">
 

### Expected behavior 🤔

As an assistive technology user, I would like to be able to access the menu offering export options (print, CSV) upon selecting the Export button using the keyboard only. 

### Steps to reproduce 🕹

CodeSandbox Demo: https://codesandbox.io/s/1jr8ds?file=/demo.js

Steps:

1. Turn the screen reader on or use the Tab key to navigate to the Export button
2. When pressing the Export button, the menu with exporting options appear, but keyboard-only users can't access it because of the tabindex="-1" on it.

### Context 🔦

Assistive technology users should have access to all features within the MUI Data Grid component. Downloading CSV, which is one of the export options, is a very important feature of our application and it is inaccessible for AT users. Developers on our team need to develop a custom CSV download feature to provide this option to all users.

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>
  System:
    OS: macOS 12.4
  Binaries:
    Node: 18.4.0 - /opt/homebrew/bin/node
    Yarn: Not Found
    npm: 8.12.1 - /opt/homebrew/bin/npm
  Browsers:
    Chrome: 103.0.5060.114
    Edge: Not Found
    Firefox: 101.0.1
    Safari: 15.5
  npmPackages:
    react:  18.2.0 
    react-dom:  18.2.0 
</details>

**Comment by m4theushw** [comment]
We missed to spread the remaining props to the `MenuItem` component. The `MenuList` sets `autoFocus` on the first element, but since this prop is not forwarded to the root element, the menu becomes inaccessible. Do you want to open a PR with the diff below?

```diff
diff --git a/packages/grid/x-data-grid/src/components/toolbar/GridToolbarExport.tsx b/packages/grid/x-data-grid/src/components/toolbar/GridToolbarExport.tsx
index b1227a06a..d3f7e69a3 100644
--- a/packages/grid/x-data-grid/src/components/toolbar/GridToolbarExport.tsx
+++ b/packages/grid/x-data-grid/src/components/toolbar/GridToolbarExport.tsx
@@ -31,7 +31,7 @@ export interface GridToolbarExportProps extends ButtonProps {
 
 export const GridCsvExportMenuItem = (props: GridCsvExportMenuItemProps) => {
   const apiRef = useGridApiContext();
-  const { hideMenu, options } = props;
+  const { hideMenu, options, ...other } = props;
 
   return (
     <MenuItem
@@ -39,6 +39,7 @@ export const GridCsvExportMenuItem = (props: GridCsvExportMenuItemProps) => {
         apiRef.current.exportDataAsCsv(options);
         hideMenu?.();
       }}
+      {...other}
     >
       {apiRef.current.getLocaleText('toolbarExportCSV')}
     </MenuItem>
@@ -47,7 +48,7 @@ export const GridCsvExportMenuItem = (props: GridCsvExportMenuItemProps) => {
 
 export const GridPrintExportMenuItem = (props: GridPrintExportMenuItemProps) => {
   const apiRef = useGridApiContext();
-  const { hideMenu, options } = props;
+  const { hideMenu, options, ...other } = props;
 
   return (
     <MenuItem
@@ -55,6 +56,7 @@ export const GridPrintExportMenuItem = (props: GridPrintExportMenuItemProps) =>
         apiRef.current.exportDataAsPrint(options);
         hideMenu?.();
       }}
+      {...other}
     >
       {apiRef.current.getLocaleText('toolbarExportPrint')}
     </MenuItem>
```

The same should be applied to the Excel button too.

```diff
diff --git a/packages/grid/x-data-grid-premium/src/components/GridExcelExportMenuItem.tsx b/packages/grid/x-data-grid-premium/src/components/GridExcelExportMenuItem.tsx
index 8a1fa0b1a..79e6e187b 100644
--- a/packages/grid/x-data-grid-premium/src/components/GridExcelExportMenuItem.tsx
+++ b/packages/grid/x-data-grid-premium/src/components/GridExcelExportMenuItem.tsx
@@ -9,7 +9,7 @@ export type GridExcelExportMenuItemProps = GridExportMenuItemProps<GridExcelExpo
 
 const GridExcelExportMenuItem = (props: GridExcelExportMenuItemProps) => {
   const apiRef = useGridApiContext();
-  const { hideMenu, options } = props;
+  const { hideMenu, options, ...other } = props;
 
   return (
     <MenuItem
@@ -17,6 +17,7 @@ const GridExcelExportMenuItem = (props: GridExcelExportMenuItemProps) => {
         apiRef.current.exportDataAsExcel(options);
         hideMenu?.();
       }}
+      {...other}
     >
       {apiRef.current.getLocaleText('toolbarExportExcel')}
     </MenuItem>
```

In the column menu we use a different approach. The focus in on the menu, not the first item, but keyboard still works. The approach from the export menu is the same from https://mui.com/material-ui/react-menu/#menulist-composition

**Comment by nogalpaulina** [comment]
Thank you @m4theushw for reviewing my issue. I opened a [PR with the diff above](https://github.com/mui/mui-x/pull/5486).

**Comment by m4theushw** [comment]
Thanks for submitting the fix. It will be included in the release of next week. If you don't want to wait for it, you can use the packages generated by your PR: https://ci.codesandbox.io/status/mui/mui-x/pr/5486/builds/270763

**Comment by nogalpaulina** [comment]
Thanks, @m4theushw!
